app/authPopup.js

```diff
@@ -2,62 +2,80 @@
 // configuration parameters are located at authConfig.js
 const myMSALObj = new msal.PublicClientApplication(msalConfig);
 
+let username = "";
+
+function loadPage() {
+    // need to call getAccount here?
+    const currentAccounts = myMSALObj.getAllAccounts();
+    if (currentAccounts === null) {
+        return;
+    } else if (currentAccounts.length > 1) {
+        // Add choose account code here
+        console.warn("Multiple accounts detected.");
+    } else if (currentAccounts.length === 1) {
+        username = currentAccounts[0].username;
+        showWelcomeMessage(currentAccounts[0]);
+    }
+}
+
+function handleResponse(resp) {
+    if (resp !== null) {
+        username = resp.account.username;
+        showWelcomeMessage(resp.account);
+    } else {
+        loadPage();
+    }
+}
+
 function signIn() {
-    myMSALObj.loginPopup(loginRequest)
-        .then(loginResponse => {
-            console.log('id_token acquired at: ' + new Date().toString());
-            console.log(loginResponse);
-
-            if (myMSALObj.getAccount()) {
-                showWelcomeMessage(myMSALObj.getAccount());
-            }
-        }).catch(error => {
-            console.error(error);
-        });
+    myMSALObj.loginPopup(loginRequest).then(handleResponse).catch(error => {
+        console.error(error);
+    });
 }
 
 function signOut() {
-    myMSALObj.logout();
+    const logoutRequest = {
+        account: myMSALObj.getAccountByUsername(username)
+    };
+
+    myMSALObj.logout(logoutRequest);
 }
 
 function getTokenPopup(request) {
-    return myMSALObj.acquireTokenSilent(request)
-        .catch(error => {
-            console.warn(error);
-            console.warn("silent token acquisition fails. acquiring token using popup");
-
+    request.account = myMSALObj.getAccountByUsername(username);
```

Comment:
if accounts have the same the "username" in some use cases, you may need to add guidance to call `getAccountsByFilter()`(which may not be a public API right now). At the minimum, have docs to explain what all consists of accounts and how to identify one before sending it to a silent call. This is applicable for `msal-node` too.

Add a comment which points to the doc and that will be good. I can help if needed later in the day.

Reply:
done.